### PR TITLE
chore(weave): Minimize dependencies between Trace and Query code bases

### PR DESCRIPTION
**Pull request by andrewtruong**
## Description

This PR removes most of the dependencies between the two packages and duplicates code where necessary.  The remaining dependencies include:
1. Legacy in `trace`:
    1. The `op` decorator in `op.py`
    2. `serve_fastapi.py`
2. Trace in `legacy`
    1. `monitoring/`

## Testing

How was this PR tested?

### CODE REVIEW

File: weave/context_state.py [hunk 1]
```diff
@@ -1,6 +1,7 @@
 import contextlib
```
Comment (tssweeney) [inline weave/context_state.py:1]: Should we note in the header that this is a "legacy" file? `WandbApiContext` is a query service concept

File: weave/trace/env.py [hunk 2]
```diff
@@ -1,7 +1,142 @@
...
+
+def weave_trace_server_url() -> str:
+    base_url = wandb_frontend_base_url()
+    default = "https://trace.wandb.ai"
+    if base_url != "https://api.wandb.ai":
+        default = base_url + "/traces"
+    return os.getenv("WF_TRACE_SERVER_URL", default)
+
+
+def _wandb_api_key_via_env() -> Optional[str]:
```
Comment (tssweeney) [inline weave/trace/env.py:73]: this is all query service stuff.

Comment (tssweeney) [reply weave/trace/env.py:73]: all the `wandb_api_*` functions

File: weave/trace/errors.py [hunk 3]
```diff
@@ -23,3 +23,6 @@ class WeaveDefinitionError(Error): ...
 
 
 class WeaveWandbAuthenticationException(Error): ...
+
+
+class WeaveConfigurationError(Error): ...
```
Comment (tssweeney) [inline weave/trace/errors.py:28]: legacy

File: weave/legacy/weave/refs_internal.py [hunk 4]
```diff
@@ -0,0 +1,215 @@
+# This file contains the definition for "Internal" weave refs. These should never be
```
Comment (tssweeney) [inline weave/legacy/weave/refs_internal.py:1]: This is not legacy!

File: weave/legacy/weave/ref_util.py [hunk 5]
```diff
@@ -0,0 +1,106 @@
+import dataclasses
```
Comment (tssweeney) [inline weave/legacy/weave/ref_util.py:1]: not legacy

File: weave/wandb_interface/wandb_api.py [hunk 6]
```diff
@@ -0,0 +1,449 @@
+# Official interface for interacting with the W&B API. All
```
Comment (tssweeney) [inline weave/wandb_interface/wandb_api.py:1]: This is all legacy

File: weave/context_state.py (hunk 1 above)
Comment (andrewtruong) [reply weave/context_state.py:1]: yeah, it's moved in the next pr:
https://github.com/wandb/weave/pull/2471

File: weave/trace/env.py (hunk 2 above)
Comment (andrewtruong) [reply weave/trace/env.py:73]: yeah, it's moved in the next pr:
https://github.com/wandb/weave/pull/2471

File: weave/trace/errors.py (hunk 3 above)
Comment (andrewtruong) [reply weave/trace/errors.py:28]: yeah, it's moved in the next pr:
https://github.com/wandb/weave/pull/2471

File: weave/legacy/weave/refs_internal.py (hunk 4 above)
Comment (andrewtruong) [reply weave/legacy/weave/refs_internal.py:1]: I think there is some dependency on the refs code in query service.  There is a duplicate file at `trace/refs_internal.py` for the trace code.

File: weave/legacy/weave/ref_util.py (hunk 5 above)
Comment (andrewtruong) [reply weave/legacy/weave/ref_util.py:1]: I think there is some dependency on the refs code in query service. There is a duplicate file at `trace/ref_util.py` for the trace code.

File: weave/wandb_interface/wandb_api.py (hunk 6 above)
Comment (andrewtruong) [reply weave/wandb_interface/wandb_api.py:1]: yeah, it's moved in the next pr:
https://github.com/wandb/weave/pull/2471

**Review by tssweeney — APPROVED** [review]

**Pull request merged**